ocaml/xapi/repository_helpers.ml

```diff
@@ -231,6 +231,17 @@ let assert_gpgkey_path_is_valid path =
     raise Api_errors.(Server_error (invalid_gpgkey_path, [path]))
   )
 
+let assert_remote_pool_url_is_valid ~url =
+  let uri = Uri.of_string url in
+  match (Uri.scheme uri, Uri.host uri, Uri.path uri) with
+  | Some "https", Some host, path
+    when path = Constants.get_enabled_repository_uri
+         && Helpers.is_valid_ip `ipv4or6 host ->
+      ()
+  | _ ->
+      error "Invalid url %s" url ;
```

Comment:
Will be helpful to also log the expected format: "https://<coordinator-ip>/repository/enabled"